src/bin/pg_autoctl/cli_do_fsm.c

```diff
@@ -320,13 +320,6 @@ cli_do_fsm_assign(int argc, char **argv)
 	/* assign the new state */
 	keeper.state.assigned_role = goalState;
 
-	/* roll the state machine */
-	if (!keeper_fsm_reach_assigned_state(&keeper))
```

Comment:
With this removal, how does the CLI make use of `keeper.otherNodes` as assigned in the 4-argument version of `do fsm assign`?

Reply:
I was not sure if we needed to expose the node list when the monitor is disabled, but if we want to manage HBA and more importantly the replication slots, then it's fair to say that we need to maintain the list of other nodes.

What I think should happen now is that we implement a set of commands to manage the list of other nodes, basically a text file containing a single array, as in the following example. The monitor replacement would have to use the new command to provide this JSON list in a file that the background process can read on every loop, and react to changes exactly like when processing the result of `pgautofailover.get_other_nodes` on the monitor.

Then I will also implement a new test file for the `--disable-monitor` option, and we will be able to add the case you found in there, either in this PR or another one.

```
$ PG_AUTOCTL_DEBUG=1 pg_autoctl do monitor get others --pgdata ./node1/ --json
[
    {
        "node_id": 2,
        "node_lsn": "0/4000110",
        "node_host": "localhost",
        "node_name": "node2",
        "node_port": 5502,
        "node_is_primary": false
    },
    {
        "node_id": 3,
        "node_lsn": "0/4000110",
        "node_host": "localhost",
        "node_name": "node3",
        "node_port": 5503,
        "node_is_primary": false
    },
    {
        "node_id": 4,
        "node_lsn": "0/4000110",
        "node_host": "localhost",
        "node_name": "node4",
        "node_port": 5504,
        "node_is_primary": false
    }
]

```

Reply:
Sounds reasonable.

Reply:
Added support for the following new commands, and the main service keeper loop now reads the file at each loop too:

```
  pg_autoctl do fsm nodes
    get  Get the list of nodes from file (see --disable-monitor)
    set  Set the list of nodes to file (see --disable-monitor)
```

Next up, unit tests using `--disable-monitor`.

Reply:
And unit testing of `--disable-monitor` is in, with many fixes to make it possible.